packages/eslint-plugin/src/rules/utils/createRule.ts

```diff
@@ -16,10 +16,7 @@
 
 import { ESLintUtils } from "@typescript-eslint/utils";
 
-// N.B. We need an explicit type annotation because of the following error:
-// The inferred type of 'createRule' cannot be named without a reference to '../../../../../node_modules/@typescript-eslint/utils/dist/ts-eslint/Rule'. This is likely not portable. A type annotation is necessary.ts(2742)
-
 /** Create a rule and automatically fill its url based on the rule name. */
-export const createRule: ReturnType<typeof ESLintUtils.RuleCreator> = ESLintUtils.RuleCreator(
```

Comment:
I was able to remove this explicit typedef (added in #6339) due to the fix in typescript-eslint here: https://github.com/typescript-eslint/typescript-eslint/issues/7605

... but all the other explicit typdefs added in #6339, like those in `packages/eslint-plugin/src/rules/no-deprecated-components/`, need to stay :(